### ISSUE TITLE
NPEs in v2.7.0

### ISSUE THREAD

**Issue by misaochan** [commons-app/apps-android-commons#1427]
New NPEs in v 2.7.0, from the Play Store dev console:

Today, 19:02 on app version 83
Samsung Galaxy A5(2017) (a5y17lte), 3072MB RAM, Android 7.0
Report 1 of 1

```
java.lang.NullPointerException: 
  at fr.free.nrw.commons.upload.UploadController.startUpload (UploadController.java:99)
  at fr.free.nrw.commons.upload.ShareActivity.uploadBegins (ShareActivity.java:168)
  at fr.free.nrw.commons.upload.ShareActivity.uploadActionInitiated (ShareActivity.java:139)
  at fr.free.nrw.commons.upload.SingleUploadFragment.onOptionsItemSelected (SingleUploadFragment.java:94)
  at android.support.v4.app.Fragment.performOptionsItemSelected (Fragment.java:2474)
  at android.support.v4.app.FragmentManagerImpl.dispatchOptionsItemSelected (FragmentManager.java:3307)
  at android.support.v4.app.FragmentController.dispatchOptionsItemSelected (FragmentController.java:344)
  at android.support.v4.app.FragmentActivity.onMenuItemSelected (FragmentActivity.java:367)
  at android.support.v7.app.AppCompatActivity.onMenuItemSelected (AppCompatActivity.java:195)
  at android.support.v7.view.WindowCallbackWrapper.onMenuItemSelected (WindowCallbackWrapper.java:108)
  at android.support.v7.view.WindowCallbackWrapper.onMenuItemSelected (WindowCallbackWrapper.java:108)
  at android.support.v7.app.ToolbarActionBar$2.onMenuItemClick (ToolbarActionBar.java:65)
  at android.support.v7.widget.Toolbar$1.onMenuItemClick (Toolbar.java:202)
  at android.support.v7.widget.ActionMenuView$MenuBuilderCallback.onMenuItemSelected (ActionMenuView.java:780)
  at android.support.v7.view.menu.MenuBuilder.dispatchMenuItemSelected (MenuBuilder.java:822)
  at android.support.v7.view.menu.MenuItemImpl.invoke (MenuItemImpl.java:171)
  at android.support.v7.view.menu.MenuBuilder.performItemAction (MenuBuilder.java:973)
  at android.support.v7.view.menu.MenuBuilder.performItemAction (MenuBuilder.java:963)
  at android.support.v7.widget.ActionMenuView.invokeItem (ActionMenuView.java:624)
  at android.support.v7.view.menu.ActionMenuItemView.onClick (ActionMenuItemView.java:150)
  at android.view.View.performClick (View.java:6261)
  at android.widget.TextView.performClick (TextView.java:11185)
  at android.view.View$PerformClick.run (View.java:23752)
  at android.os.Handler.handleCallback (Handler.java:751)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:154)
  at android.app.ActivityThread.main (ActivityThread.java:6776)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1518)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1408)
```

Yesterday, 18:00 on app version 83
Huawei M3 (hwbeethoven), 3072MB RAM, Android 7.0
Report 1 of 1

```
java.lang.NullPointerException: 
  at android.support.design.widget.BottomSheetBehavior.onInterceptTouchEvent (BottomSheetBehavior.java:299)
  at android.support.design.widget.CoordinatorLayout.performIntercept (CoordinatorLayout.java:460)
  at android.support.design.widget.CoordinatorLayout.onInterceptTouchEvent (CoordinatorLayout.java:499)
  at android.view.ViewGroup.dispatchTouchEvent (ViewGroup.java:2212)
  at android.view.ViewGroup.dispatchTransformedTouchEvent (ViewGroup.java:2671)
  at android.view.ViewGroup.dispatchTouchEvent (ViewGroup.java:2301)
  at android.view.ViewGroup.dispatchTransformedTouchEvent (ViewGroup.java:2671)
  at android.view.ViewGroup.dispatchTouchEvent (ViewGroup.java:2301)
  at android.view.ViewGroup.dispatchTransformedTouchEvent (ViewGroup.java:2671)
  at android.view.ViewGroup.dispatchTouchEvent (ViewGroup.java:2301)
  at android.view.ViewGroup.dispatchTransformedTouchEvent (ViewGroup.java:2671)
  at android.view.ViewGroup.dispatchTouchEvent (ViewGroup.java:2301)
  at android.view.ViewGroup.dispatchTransformedTouchEvent (ViewGroup.java:2671)
  at android.view.ViewGroup.dispatchTouchEvent (ViewGroup.java:2301)
  at android.view.ViewGroup.dispatchTransformedTouchEvent (ViewGroup.java:2671)
  at android.view.ViewGroup.dispatchTouchEvent (ViewGroup.java:2301)
  at com.android.internal.policy.DecorView.superDispatchTouchEvent (DecorView.java:447)
  at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent (PhoneWindow.java:1871)
  at android.app.Activity.dispatchTouchEvent (Activity.java:3213)
  at android.support.v7.view.WindowCallbackWrapper.dispatchTouchEvent (WindowCallbackWrapper.java:68)
  at android.support.v7.view.WindowCallbackWrapper.dispatchTouchEvent (WindowCallbackWrapper.java:68)
  at com.android.internal.policy.DecorView.dispatchTouchEvent (DecorView.java:409)
  at android.view.View.dispatchPointerEvent (View.java:10232)
  at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent (ViewRootImpl.java:4856)
  at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess (ViewRootImpl.java:4717)
  at android.view.ViewRootImpl$InputStage.deliver (ViewRootImpl.java:4249)
  at android.view.ViewRootImpl$InputStage.onDeliverToNext (ViewRootImpl.java:4302)
  at android.view.ViewRootImpl$InputStage.forward (ViewRootImpl.java:4268)
  at android.view.ViewRootImpl$AsyncInputStage.forward (ViewRootImpl.java:4395)
  at android.view.ViewRootImpl$InputStage.apply (ViewRootImpl.java:4276)
  at android.view.ViewRootImpl$AsyncInputStage.apply (ViewRootImpl.java:4452)
  at android.view.ViewRootImpl$InputStage.deliver (ViewRootImpl.java:4249)
  at android.view.ViewRootImpl$InputStage.onDeliverToNext (ViewRootImpl.java:4302)
  at android.view.ViewRootImpl$InputStage.forward (ViewRootImpl.java:4268)
  at android.view.ViewRootImpl$InputStage.apply (ViewRootImpl.java:4276)
  at android.view.ViewRootImpl$InputStage.deliver (ViewRootImpl.java:4249)
  at android.view.ViewRootImpl.deliverInputEvent (ViewRootImpl.java:6676)
  at android.view.ViewRootImpl.doProcessInputEvents (ViewRootImpl.java:6650)
  at android.view.ViewRootImpl.enqueueInputEvent (ViewRootImpl.java:6611)
  at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent (ViewRootImpl.java:6803)
  at android.view.InputEventReceiver.dispatchInputEvent (InputEventReceiver.java:192)
  at android.os.MessageQueue.nativePollOnce (Native Method)
  at android.os.MessageQueue.next (MessageQueue.java:356)
  at android.os.Looper.loop (Looper.java:138)
  at android.app.ActivityThread.main (ActivityThread.java:6523)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:942)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:832)
```

**Comment by neslihanturan** [comment]
@misaochan the first NPE is related with this line:
`                null, null, sessionManager.getCurrentAccount().name,`
seems like after our update, user is not logged in anymore, so this crash must be related with "there is no current account and its name problem". I think this will be solved after #1393 is solved.

**Comment by neslihanturan** [comment]
And maybe second one is a known bug https://github.com/material-components/material-components-android/issues/43

**Comment by misaochan** [comment]
I agree re: the first NPE. Once #1393 is solved I guess we can try and see if the first NPE disappears.

The second one is interesting. Doesn't seem like there's anything we can do to solve it in that case?

**Comment by domdomegg** [comment]
Think first NPE fixed in #1454, second in https://github.com/material-components/material-components-android/commit/24c40ffa2aa0ba9ea33a3dd9755be050c49e4da0
